cishouseholds/derive.py

```diff
@@ -1184,6 +1184,8 @@ def assign_column_to_date_string(
     -------
     pyspark.sql.DataFrame
     """
+    if reference_column not in df.columns:
```

Comment:
Please could this be moved outside of the function and used for these specific date string function calls only? We generally want functions only to be used when the relevant columns exist, but have to make an exception for backwards compatibility here. We can always drop the `if` parts once the digital code is in production